### PR TITLE
Add slots to hostfile

### PR DESCRIPTION
**Pull request by tenzen-y**
Signed-off-by: Yuki Iwai <yuki.iwai.tz@gmail.com>

I added slots to the hostfile for the Horovod.

ref: https://horovod.readthedocs.io/en/stable/running_include.html#run-horovod

Fixes: #445

### CODE REVIEW

File: pkg/controller/mpi_job_controller.go [hunk 1]
```diff
@@ -1182,8 +1182,16 @@ func (c *MPIJobController) doUpdateJobStatus(mpiJob *kubeflow.MPIJob) error {
...
+	if mpiJob.Spec.SlotsPerWorker != nil {
+		slots = int(*mpiJob.Spec.SlotsPerWorker)
+	}
 	for i := 0; i < int(workerReplicas); i++ {
-		buffer.WriteString(fmt.Sprintf("%s%s-%d.%s.%s.svc\n", mpiJob.Name, workerSuffix, i, workersService, mpiJob.Namespace))
+		if mpiJob.Spec.MPIImplementation == kubeflow.MPIImplementationOpenMPI {
+			buffer.WriteString(fmt.Sprintf("%s%s-%d.%s.%s.svc slots=%d\n", mpiJob.Name, workerSuffix, i, workersService, mpiJob.Namespace, slots))
+		} else if mpiJob.Spec.MPIImplementation == kubeflow.MPIImplementationIntel {
+			buffer.WriteString(fmt.Sprintf("%s%s-%d.%s.%s.svc:%d\n", mpiJob.Name, workerSuffix, i, workersService, mpiJob.Namespace, slots))
+		}
```
Comment (tenzen-y) [inline pkg/controller/mpi_job_controller.go:1194]: OpenMPI and IntelMPI have different host file formats.

- OpenMPI: https://www.open-mpi.org/faq/?category=running#mpirun-scheduling
- IntelMPI: https://www.intel.com/content/www/us/en/developer/articles/technical/controlling-process-placement-with-the-intel-mpi-library.html

File: pkg/controller/mpi_job_controller.go [hunk 2]
```diff
@@ -1182,8 +1182,16 @@ func (c *MPIJobController) doUpdateJobStatus(mpiJob *kubeflow.MPIJob) error {
 func newConfigMap(mpiJob *kubeflow.MPIJob, workerReplicas int32) *corev1.ConfigMap {
 	var buffer bytes.Buffer
 	workersService := mpiJob.Name + workerSuffix
+	slots := 1
+	if mpiJob.Spec.SlotsPerWorker != nil {
+		slots = int(*mpiJob.Spec.SlotsPerWorker)
+	}
 	for i := 0; i < int(workerReplicas); i++ {
-		buffer.WriteString(fmt.Sprintf("%s%s-%d.%s.%s.svc\n", mpiJob.Name, workerSuffix, i, workersService, mpiJob.Namespace))
+		if mpiJob.Spec.MPIImplementation == kubeflow.MPIImplementationOpenMPI {
```
Comment (alculquicondor) [inline pkg/controller/mpi_job_controller.go:1190]: nit: use a switch statement for behavior that varies per enums.

**Review by alculquicondor — COMMENTED** [review]
/approve

File: pkg/controller/mpi_job_controller.go (hunk 2 above)
Comment (tenzen-y) [reply pkg/controller/mpi_job_controller.go:1190]: Makes sense.

Comment (tenzen-y) [reply pkg/controller/mpi_job_controller.go]: Done.

**Comment by tenzen-y** [comment]
@alculquicondor I have addressed your comments and squashed into one. PTAL.

**Pull request merged**